docs/concepts/executor/docarray-v2.md

```diff
@@ -0,0 +1,122 @@
+the version that is currently used in Jina, the `Document` as a fixed schema and the Executor performs in place operations
+on directly on it. With DocArray v2 things change slightly. Each executor will need to define its own input schema
+and output schema. Of course, we still provided predefined schema. We believe this will allow Executor to be more
+expressive and flexible.
+
+## (Beta) New Executor API
+
+To reflect the change with DocArray v2, the Executor API now supports schema definition. The 
+design is freely inspired by [FastAPI](https://fastapi.tiangolo.com/). 
+
+
+```{code-block} python
+---
+emphasize-lines: 17,18
+---
+from jina import Executor, requests
+from docarray import BaseDocument, DocumentArray
+from docarray.documents import Image
+from docarray.typing import AnyTensor
+
+import numpy as np
+
+class InputDoc(BaseDocument):
+    img: Image
+
+class OutputDoc(BaseDocument):
+    embedding: AnyTensor
+
+class MyExec(Executor):
+    @requests(on='/bar')
+    def bar(
+        self, docs: DocumentArray[InputDoc], **kwargs
+    ) -> DocumentArray[OutputDoc]:
+        docs_return = DocumentArray[OutputDoc](
+            [OutputDoc(embedding=np.zeros((100, 1))) for _ in range(len(docs))]
+        )
+        return docs_return
+```
+
+For our Executor we define an input schema `InputDoc` and an output schema `OutputDoc` which are `Document`. 
+We then define the `bar` endpoint which takes as input a `DocumentArray` of `InputDoc` and return a `DocumentArray` of
+`OutputDoc`. Note that here the type hint is actually more that just a hint, and like in FastAPI we infer the actual
+schema of the endpoint from the type hint.
+
+There is also a way to explicitly define the schema of the endpoint. This is done by using the `input_type` and
+`output_type` parameters of the `requests` decorator.
+
+
+```{code-block} python
+---
+emphasize-lines: 4,5
+---
+class MyExec(Executor):
+    @requests(
+        on='/bar',
+        input_type=DocumentArray[InputDoc],
+        output_type=DocumentArray[OutputDoc],
+    )
+    def bar(self, docs, **kwargs) 
+        docs_return = DocumentArray[OutputDoc](
+            [OutputDoc(embedding=np.zeros((100, 1))) for _ in range(len(docs))]
+        )
+        return docs_return
+```
+
+If there is no `input_type` and `output_type` the type hint is used to infer the schema. If there are both `input_type`
+and `output_type` will be used.
+
+
+## (Beta) Client API
+
+In the client, you similarly specify the schema that is expected to be returned by the Flow. You can pass the return type by using the `return_type` parameter of the `client.post`
+
+```{code-block} python
+---
+emphasize-lines: 7
+---
+from jina import Deployment
+
+with Deployment(uses=MyExec) as dep:
+    docs = dep.post(
+        on='/bar',
+        inputs=InputDoc(img=Image(tensor=np.zeros((3, 224, 224)))),
+        return_type=DocumentArray[OutputDoc],
```

Comment:
is it necessary to always specify `DocumentArray[OutputDoc]` ?
I believe `OutputDoc` is enough